controls/C-0012/policy.yaml

```diff
@@ -0,0 +1,60 @@
+  paramKind:
+    apiVersion: kubescape.io/v1
+    kind: ControlConfiguration
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   [""]
+      apiVersions: ["v1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["pods"]
+    - apiGroups:   ["apps"]
+      apiVersions: ["v1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["deployments","replicasets","daemonsets","statefulsets"]
+    - apiGroups:   ["batch"]
+      apiVersions: ["v1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["jobs","cronjobs"]
+    - apiGroups:   [""]
+      apiVersions: ["v1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["configmaps"]
+  validations:
+    - expression: > 
+        object.kind != 'Pod' || object.spec.containers.all(container, !has(container.env) || container.env.all(envVariable, 
+        !params.settings.sensitiveKeyNames.exists(sensitiveKey, envVariable.name.lowerAscii().contains(sensitiveKey)) ||
+        !has(envVariable.value) || (envVariable.value == "") ||
+        params.settings.sensitiveValuesAllowed.exists(allowedVal, envVariable.value == allowedVal)
+        ))
+      message: "Pods has one or more containers with sensitive information in environment variables! (see more at https://hub.armosec.io/docs/c-0012)"
+    - expression: > 
+        ['Deployment','ReplicaSet','DaemonSet','StatefulSet','Job'].all(kind, object.kind != kind) || object.spec.template.spec.containers.all(container, !has(container.env) || container.env.all(envVariable, 
+        !params.settings.sensitiveKeyNames.exists(sensitiveKey, envVariable.name.lowerAscii().contains(sensitiveKey)) ||
+        !has(envVariable.value) || (envVariable.value == "") ||
+        params.settings.sensitiveValuesAllowed.exists(allowedVal, envVariable.value == allowedVal)
+        ))
+      message: "Workloads has one or more containers with sensitive information in environment variables! (see more at https://hub.armosec.io/docs/c-0012)"
+    - expression: > 
+        object.kind != 'CronJob' || object.spec.jobTemplate.spec.containers.all(container, !has(container.env) || container.env.all(envVariable, 
+        !params.settings.sensitiveKeyNames.exists(sensitiveKey, envVariable.name.lowerAscii().contains(sensitiveKey)) ||
+        !has(envVariable.value) || (envVariable.value == "") ||
+        params.settings.sensitiveValuesAllowed.exists(allowedVal, envVariable.value == allowedVal)
+        ))
+      message: "CronJob has one or more containers with sensitive information in environment variables! (see more at https://hub.armosec.io/docs/c-0012)"
+    - expression: > 
+        object.kind != 'ConfigMap' || object.data.all(key,
+        (
+          (
+            !params.settings.sensitiveKeyNames.exists(sensitiveKey, key.lowerAscii().contains(sensitiveKey)) &&
+            !params.settings.sensitiveValues.exists(sensitiveVal, key.lowerAscii().matches(sensitiveVal))
+          ) ||
+          object.data.key == "" ||
```

Comment:
Hi @slashben, I want to access the variable `key` that I have defined in `object.data.all`, but CEL is taking it literally and searching for `"key"`. 

I searched for a way to access map values inside `all` macros but to no avail. That is the only thing blocking the completion of this control.

Reply:
This is not good:
```
object.data.key == ""
```
I think you meant
```
key == ""
```

Reply:
* Hi Ben, I have been looking to access the value in the map corresponding to that key. 
* I have to use `[]` to do so. I updated and it is working as expected.